nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/encrypt/PasswordBasedCipherPropertyEncryptorTest.java

```diff
@@ -0,0 +1,89 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.encrypt;
+
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.nifi.security.util.EncryptionMethod;
+import org.apache.nifi.security.util.crypto.PBECipherProvider;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+public class PasswordBasedCipherPropertyEncryptorTest {
+    private static final Charset DEFAULT_CHARSET = StandardCharsets.UTF_8;
+
+    private static final String PROPERTY = String.class.getName();
+
+    private static final int ENCRYPTED_BINARY_LENGTH = 48;
+
+    @SuppressWarnings("deprecation")
+    private static final PBECipherProvider CIPHER_PROVIDER = new org.apache.nifi.security.util.crypto.NiFiLegacyCipherProvider();
+
+    private static final EncryptionMethod ENCRYPTION_METHOD = EncryptionMethod.MD5_256AES;
+
+    private static final String PASSWORD = String.class.getName();
+
+    private PasswordBasedCipherPropertyEncryptor encryptor;
+
+    @Before
+    public void setUp() {
+        encryptor = new PasswordBasedCipherPropertyEncryptor(CIPHER_PROVIDER, ENCRYPTION_METHOD, PASSWORD);
+    }
+
+    @Test
+    public void testEncryptDecrypt() {
+        final String encrypted = encryptor.encrypt(PROPERTY);
+        final String decrypted = encryptor.decrypt(encrypted);
+        assertEquals(PROPERTY, decrypted);
+    }
+
+    @Test
+    public void testEncryptHexadecimalEncoded() throws DecoderException {
+        final String encrypted = encryptor.encrypt(PROPERTY);
+        final byte[] decoded = Hex.decodeHex(encrypted);
+        assertEquals(ENCRYPTED_BINARY_LENGTH, decoded.length);
+    }
+
+    @Test
+    public void testDecryptEncryptionException() {
+        final String encodedProperty = Hex.encodeHexString(PROPERTY.getBytes(DEFAULT_CHARSET));
+        assertThrows(EncryptionException.class, () -> encryptor.decrypt(encodedProperty));
+    }
+
+    @Test
+    public void testEqualsHashCode() {
+        final PasswordBasedCipherPropertyEncryptor equivalentEncryptor = new PasswordBasedCipherPropertyEncryptor(CIPHER_PROVIDER, ENCRYPTION_METHOD, PASSWORD);
+        assertTrue(encryptor.equals(equivalentEncryptor));
+        assertEquals(encryptor.hashCode(), equivalentEncryptor.hashCode());
+    }
+
+    @Test
+    public void testEqualsHashCodeDifferentPassword() {
+        final PasswordBasedCipherPropertyEncryptor differentEncryptor = new PasswordBasedCipherPropertyEncryptor(CIPHER_PROVIDER, ENCRYPTION_METHOD, String.class.getSimpleName());
+        assertFalse(encryptor.equals(differentEncryptor));
```

Comment:
Optionally, this could be simplified to assertNotEquals.

Reply:
Will make the change as described.